compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -9070,14 +9073,15 @@ private STNode parseResourceAccessSegmentRhs(boolean isRhsExpr, boolean isInMatc
         }
     }
     
-    private boolean isEndOfResourceAccessPathSegments(SyntaxKind nextTokenKind,
+    private boolean isEndOfResourceAccessPathSegments(STToken nextToken,
                                                       boolean isRhsExpr, boolean isInMatchGuard) {
+        SyntaxKind nextTokenKind = nextToken.kind;
         switch (nextTokenKind) {
```

Comment:
Since there is only one usage, no need to define a separate variable.
```suggestion
        switch (nextToken.kind) {
```